### DISCUSSION TITLE
Tongsheng TSDZ8

### DISCUSSION THREAD

**Discussion by kb79000** [danielnilsson9/bbs-fw#214] [Q&A]
Hello, Someone has already tested the firmware on this TSDZ8 engine ?
![-8115430211846748341~2](https://github.com/danielnilsson9/bbs-fw/assets/46342882/7aa4c3a0-b1fa-4c6e-aae2-743edfd895e4)

Thx

**Comment by julienchavas** [comment]
I would like to know too.
But i don't know if it's possible.. i'm not sure that the tsdz8 use the same MCU than the tsdz2b.

**Comment by danielnilsson9** [comment]
Firmware is very unlikely to work without modifications on TSDZ8. But that is an interesting motor, will probably consider putting that on my commuter bike when my TSDZ2 is worn out.

I'm not able to find the controller for sale as a spare part, will order if I find it and do some initial investigations to see if it is possible to support.

**Reply by kb79000** [reply]
👍🏻 It looks like a good improvement actually, thank you for your answer @danielnilsson9 😉

**Reply by julienchavas** [reply]
Hello @danielnilsson9  ! 
As always, i was lost on aliexpress, and i think i found it :)

https://a.aliexpress.com/_EHPfkRH

**Reply by kb79000** [reply]
> Hello @danielnilsson9 ! As always, i was lost on aliexpress, and i think i found it :)
> 
> https://a.aliexpress.com/_EHPfkRH

It would be great if @danielnilsson9 We create software to improve the engine because the different firmware versions published on Facebook groups is far from optimal lack of responsiveness of the torque and level sensor not well calibrated 🙏🏻

**Comment by kb79000** [comment]
Hi @danielnilsson9 Yes too bad they do not sell parts at retail.
I installed it yesterday and the torque sensor changes so much compared to my BBS02B😅
On the other hand, a firmware produced by you would be very useful because the level of assistance 1 and 2 of the engine is almost unusable without assistance.
I hope you will own it soon  🤞🏻

**Reply by geomix92** [reply]
> Hi @danielnilsson9 Yes too bad they do not sell parts at retail. I installed it yesterday and the torque sensor changes so much compared to my BBS02B😅 On the other hand, a firmware produced by you would be very useful because the level of assistance 1 and 2 of the engine is almost unusable without assistance. I hope you will own it soon 🤞🏻

Hello kb79000,
would you please share your thoughts of BBS02B vs. TSDZ8 regarding build quality and noise (I have a well lubricated BBS02B and would like to know if the TSDZ8 is louder/more disturbing)?

**Comment by danielnilsson9** [comment]
I saw it was possible to flash firmware externally on this motor just as TSDZ2 so I have ordered a TSDZ8 and will start doing some investigations!

**Reply by danielnilsson9** [reply]
Flash instructions can be found along with original firmware here:
https://ebikestuff.eu/en/content/18-tsdz8-firmware-update-using-j-link-jlink-v9-programming-kit

Seems to be using a Infineon XMC1302-0064 32bit ARM Cortex M0 MCU which looks very nice.
https://www.infineon.com/cms/en/product/microcontroller/32-bit-industrial-microcontroller-based-on-arm-cortex-m/32-bit-xmc1000-industrial-microcontroller-arm-cortex-m0/xmc1302-q040x0064-ab/

The manufacturer provides a PMSM FOC implementation for this MCU which should make things a bit easier.

**Reply by kb79000** [reply]
> Flash instructions can be found along with original firmware here: https://ebikestuff.eu/en/content/18-tsdz8-firmware-update-using-j-link-jlink-v9-programming-kit
> 
> Seems to be using a Infineon XMC1302-0064 32bit ARM Cortex M0 MCU which looks very nice. https://www.infineon.com/cms/en/product/microcontroller/32-bit-industrial-microcontroller-based-on-arm-cortex-m/32-bit-xmc1000-industrial-microcontroller-arm-cortex-m0/xmc1302-q040x0064-ab/
> 
> The manufacturer provides a PMSM FOC implementation for this MCU which should make things a bit easier.

Hi @danielnilsson9,

I'm glad to hear that you're going to try out the TSDZ8! There are some firmware shared in certain groups that address the main issue, which was the lack of assistance in modes 1 and 2.

I’m sharing with you the diagram for flashing with a JLINK box: https://s.click.aliexpress.com/e/_EHZYH1J
![FB_IMG_1729354715734](https://github.com/user-attachments/assets/bcfc9d13-9f06-4192-a0c0-fbed00c609ee)
